text/0000-ember-data-singleton-record-data.md

```diff
@@ -0,0 +1,463 @@
+}
+```
+
+</td>
+</tr>
+
+</tbody>
+</table>
+
+**Updates to RecordData Methods**
+
+<table>
+<thead>
+<tr>
+<th>Old</th>
+<th>New</th>
+</tr>
+</thead>
+<tbody>
+<tr>
+<td valign="top">
+
+```ts
+interface LegacyResourceIdentifierObject {
+  type: string;
+  id: string | null;
+  clientId: string | null;
+}
+
+interface RecordDataV1 {
+  unloadRecord(): void;
+
+  @deprecated // This RFC deprecates this method without replacement. It was not in
+  // the original RecordData RFC and after investigation is unneeded once the singleton
+  // and identifiers refactoring are completed.
+  isRecordInUse(): boolean;
+  isEmpty(): boolean;
+  isNew(): boolean;
+
+  getAttr(propertyName: string): any;
+  isAttrDirty(propertyName: string): boolean;
+  changedAttributes(): AttributesChanges;
+  hasChangedAttributes(): boolean;
+  rollbackAttributes(): void;
+
+  @deprecated // this RFC deprecates this method entirely
+  getBelongsTo(propertyName: string): { data: LegacyResourceIdentifierObject };
+  @deprecated // this RFC deprecates this method entirely
+  getHasMany(propertyName: string): { data: LegacyResourceIdentifierObject[] };
+
+  willCommit(): void;
+  didCommit(data: any): void;
+
+  @deprecated // this RFC deprecates this method entirely
+  _initRecordCreateOptions(options: object): object;
+  clientDidCreate(): void;
+
+  @deprecated // this RFC deprecates this method entirely and without replacement
+  getResourceIdentifier(): LegacyResourceIdentifierObject;
+
+  // original RecordData RFC specified setBelongsTo
+  // with the 2nd arg being LegacyResourceIdentifierObject
+  // but that was not what the resulting implementation in
+  // ember-data did
+  setDirtyBelongsTo(propertyName: string, value: RecordData | null): void;
+
+  // original RecordData RFC specified setAttribute
+  // but that was not what the resulting implementation in
+  // ember-data did
+  setDirtyAttribute(propertyName: string, value: any): void;
+
+  // original RecordData RFC specified setHasMany
+  // with the 2nd arg being LegacyResourceIdentifierObject[]
+  // but that was not what the resulting implementation in
+  // ember-data did
+  setDirtyHasMany(propertyName: string, value: RecordData[]): void;
+
+  // original RecordData RFC specified addToHasMany
+  // with the 2nd arg being LegacyResourceIdentifierObject[]
+  // but that was not what the resulting implementation in
+  // ember-data did
+  addToHasMany(propertyName: string, values: RecordData[], startIndex?: number): void;
+
+  // original RecordData RFC specified removeFromHasMany
+  // with the 2nd arg being LegacyResourceIdentifierObject[]
+  // but that was not what the resulting implementation in
+  // ember-data did
+  removeFromHasMany(propertyName: string, values: RecordData[]): void;
+
+  @deprecated // this RFC deprecates this method entirely and without replacement
+  removeFromInverseRelationships(isNew: boolean): void;
+}
+```
+
+</td>
+<td valign="top">
+
+```ts
+interface RecordDataV2 {
```

Comment:
We should add `version` here or at least somewhere in the RFC fully describe the `RecordDataV2` interface.